### PR TITLE
Redesign room search interface

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/element-hq/element-web/issues/25883
~Requires https://github.com/element-hq/compound-web/pull/196~
~Requires https://github.com/matrix-org/matrix-react-sdk/pull/12698~
Requires https://github.com/matrix-org/matrix-react-sdk/pull/12721

image

### CODE REVIEW

**Comment by americanrefugee** [comment]
A few comments:
- When clicking "Cancel", the search field should be cleared as well
- When the banner is condensed, the button "Search all rooms" should remain on one line, and the text on the left ("XX results for..." and "Open the desktop app...") should be truncated with "..." on one line (in other words, no text wrapping). [Example here](https://www.figma.com/design/gpxHFDTNK796n84r1ZxYG2/Room-Header-%26-Details-(2024)?node-id=897-42675&t=AaK0QFGDKSanmpUQ-4).
- The Clear ("x") button in the active search field should be bigger, have no background, and have a rollover state [as shown in Compound](https://www.figma.com/design/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?node-id=797-8059&t=H9Sb7qo9mdczYxEn-4)

Screenshot 2024-06-27 at 10 56 13

**Comment by americanrefugee** [comment]
Design LGTM!

However, clicking "Open the desktop app" goes to the download page on element.io... I expected it to actually open the desktop app. Is that intentional?

**Comment by t3chguy** [comment]
> However, clicking "Open the desktop app" goes to the download page on element.io... I expected it to actually open the desktop app. Is that intentional?

Yes, there's no way to detect if you have the app installed. The old text was to *Use* the desktop app, which may mean acquiring it first, this is the text I shared with you in the room but then the Figma said *open*.

image

File: playwright/snapshots/invite/invite-dialog.spec.ts/invite-dialog-room-with-user-pill-linux.png
Comment (t3chguy) [inline playwright/snapshots/invite/invite-dialog.spec.ts/invite-dialog-room-with-user-pill-linux.png:1]: the bg in the corners changes =(

File: playwright/snapshots/invite/invite-dialog.spec.ts/invite-dialog-room-without-user-linux.png
Comment (t3chguy) [inline playwright/snapshots/invite/invite-dialog.spec.ts/invite-dialog-room-without-user-linux.png:1]: the bg in the corners changes =(

**Comment by americanrefugee** [comment]
> > However, clicking "Open the desktop app" goes to the download page on element.io... I expected it to actually open the desktop app. Is that intentional?
> 
> Yes, there's no way to detect if you have the app installed. The old text was to _Use_ the desktop app, which may mean acquiring it first, this is the text I shared with you in the room but then the Figma said _open_. Most users would not be using Element Web if they already had Element Desktop installed.
> 
> image
> 
> Have updated the copy to `Get the Desktop app...`

I see. If possible, can we change "Open..." back to "Use..."? Not a blocker if you can't make this change, but I at least wanted to agree with you :)

**Comment by t3chguy** [comment]
Yes we can make the copy whatever we like.

File: src/components/views/rooms/RoomSearchAuxPanel.tsx
Comment (MidhunSureshR) [inline src/components/views/rooms/RoomSearchAuxPanel.tsx:1]: OOI what does "aux" mean here?

**Review by MidhunSureshR — APPROVED** [review]

File: src/components/views/rooms/RoomSearchAuxPanel.tsx
Comment (t3chguy) [reply src/components/views/rooms/RoomSearchAuxPanel.tsx:1]: Auxilary, it is rendered within the AuxPanel so kept the nomenclature

**Pull request merged**